### PR TITLE
Adding insert guest account and SSH key feature as per #10

### PR DESCRIPTION
**Pull request by ibaldin**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -423,6 +423,10 @@
 					<artifactId>bcprov-jdk16</artifactId>
 					<groupId>bouncycastle.bcprov-jdk16</groupId>
 				</exclusion>
+				<exclusion>
```
Comment (ibaldin) [inline pom.xml]: This fixes build problems during deploy. otherwise conflict with multiple bouncy-castle dependencies.

**Comment by ibaldin** [comment]
This has been tested to work on PSC rack that was manually modified to support proper modifySliverSSH operation.

File: pom.xml [hunk 2]
```diff
@@ -423,6 +423,10 @@
 					<artifactId>bcprov-jdk16</artifactId>
 					<groupId>bouncycastle.bcprov-jdk16</groupId>
```
Comment (hinchliff) [inline pom.xml]: I'm pretty sure this exclusion is not needed, as orca.core is not providing this artifact.  This should probably be deleted.

File: src/main/java/orca/flukes/GUIUnifiedState.java [hunk 3]
```diff
@@ -877,6 +878,39 @@ public void run() {
...
+                    NewUserDialog sshKeyDialog = new NewUserDialog(GUI.getInstance().getFrame(), "Account and SSH keys", 
+                            "Paste your public SSH key here:", 20, 50);
+                    sshKeyDialog.pack();
+                    sshKeyDialog.setVisible(true);
+                    String keys = sshKeyDialog.getSSHKeys();
+                    boolean sudo = sshKeyDialog.getSudo();
+                    String username = sshKeyDialog.getUsername();
+
+                    for(OrcaNode on: g.getVertices()) {
+                        if (on instanceof OrcaNode) {
```
Comment (hinchliff) [inline src/main/java/orca/flukes/GUIUnifiedState.java]: IntelliJ points out
>Condition 'on instanceof OrcaNode' is redundant and can be replaced with '!= null'

File: src/main/java/orca/flukes/GUIUnifiedState.java [hunk 4]
```diff
@@ -877,6 +878,39 @@ public void run() {
...
+                            // call XMLRPC proxy to insert the key into node
+                            Boolean tmpRes = OrcaSMXMLRPCProxy.getInstance().modifySliverSSH(
+                                    GUIUnifiedState.getInstance().getSliceName(), 
+                                    on.getReservationGuid(), username, sudo, Arrays.asList(keys));
```
Comment (hinchliff) [inline src/main/java/orca/flukes/GUIUnifiedState.java]: Regarding the `Arrays.asList(keys)`, IntelliJ points out:
>Reports any calls to Arrays.asList() with zero arguments or only one argument. Such calls could be replaced with either a call to Collections.singletonList() or Collections.emptyList() which will save some memory.

File: src/main/java/orca/flukes/MouseMenus.java [hunk 5]
```diff
@@ -520,6 +521,57 @@ public void setPoint(Point2D point) {
...
+	    OrcaResource node;
+	    VisualizationViewer<OrcaNode, OrcaLink> visComp;
+	    Point2D point;
+
+	    public NodeInsertSSHKeyItem(final JFrame frame) {
+	        super("Insert guest SSH key ...");
+	        this.addActionListener(new ActionListener() {
+	            public void actionPerformed(ActionEvent e) {
+	                try {
+	                    // display key dialog
```
Comment (hinchliff) [inline src/main/java/orca/flukes/MouseMenus.java]: This looks like exactly the same code as above (in GUIUnifiedState.java) ?

File: src/main/java/orca/flukes/xmlrpc/OrcaSMXMLRPCProxy.java [hunk 6]
```diff
@@ -559,6 +560,58 @@ public String sliceStatus(String sliceId)  throws Exception {
 		return result;
 	}
 
+	@SuppressWarnings("unchecked")
+	public Boolean modifySliverSSH(String sliceId, String resId, String user, boolean sudo, List<String> keys) throws Exception {
+	    assert(sliceId != null);
```
Comment (hinchliff) [inline src/main/java/orca/flukes/xmlrpc/OrcaSMXMLRPCProxy.java]: `assert()` shouldn't be used in public facing code.
[Objects.requreNonNull](https://docs.oracle.com/javase/7/docs/api/java/util/Objects.html#requireNonNull(T)) from Java 7 might be the easiest thing to use.

File: src/main/java/orca/flukes/GUIUnifiedState.java (hunk 3 above)
Comment (ibaldin) [reply src/main/java/orca/flukes/GUIUnifiedState.java]: It should probably say

for (OrcaResource on: g.getVertices()) {
    if (on instanceof OrcaNode) {
   }
}

File: src/main/java/orca/flukes/MouseMenus.java (hunk 5 above)
Comment (ibaldin) [reply src/main/java/orca/flukes/MouseMenus.java]: It is similar - in mousemenus this deals with a single node and above it deals with all nodes in a slice.

**Pull request merged**